packages/hop-node/src/watchers/SettleBondedWithdrawalWatcher.ts

```diff
@@ -86,52 +90,29 @@ class SettleBondedWithdrawalWatcher extends BaseWatcher {
         continue
       }
 
-      // if all transfer ids have been marked as settled,
-      // then mark transfer root as all settled since there is nothing to settle anymore
-      const allBondableTransfersSettled = dbTransfers.every(
-        (dbTransfer: Transfer) => {
-          // A transfer should not be settled if it is unbondable
-          return !dbTransfer.isBondable || dbTransfer.withdrawalBondSettled
-        }
-      )
+      const allBondableTransfersSettled = this.syncWatcher.getIsDbTransfersAllSettled(dbTransfers)
       if (allBondableTransfersSettled) {
-        await this.db.transferRoots.update(transferRootHash!, { // eslint-disable-line @typescript-eslint/no-non-null-assertion
-          allSettled: allBondableTransfersSettled
+        await this.db.transferRoots.update(transferRootHash, {
+          allSettled: true
         })
         continue
       }
 
       // find all unique bonders that have bonded transfers in this transfer root
       const bonderSet = new Set<string>()
       for (const dbTransfer of dbTransfers) {
-        if (!dbTransfer.withdrawalBonder) {
+        const doesBonderExist = dbTransfer?.withdrawalBonder
+        const shouldTransferBeSettled = dbTransfer?.withdrawalBondSettled === false
+        if (!doesBonderExist || !shouldTransferBeSettled) {
           continue
         }
-        bonderSet.add(dbTransfer.withdrawalBonder)
+        bonderSet.add(dbTransfer.withdrawalBonder!) // eslint-disable-line @typescript-eslint/no-non-null-assertion
       }
 
       for (const bonder of bonderSet.values()) {
-        const bonderAddress = await this.bridge.getBonderAddress()
-        if (bonder !== bonderAddress) {
-          continue
```

Comment:
is it expected that a bonder may settle another bonders unsettled transfers?

Reply:
At this time, yes. This comes into play when transactions are manually bonded.

In an ideal world, the bonder themself bonds everything and only settles their own transfers, but given some manual bonds and potential edge-case bonder swaps, I believe it makes sense to bond all at this time. Let me know if you agree with this.

Reply:
I'm good with that